### PR TITLE
set up STI for chess pieces and their sub classes (piece type)

### PR DESCRIPTION
**Pull request by abconner25**
set up the chess_piece.rb model as well as added the corresponding table to the db. Also used single table inheritance to make subclass models for each individual chess piece.

### CODE REVIEW

File: app/models/bishop.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class bishop < ChessPiece
```
Comment (talk2MeGooseman) [inline app/models/bishop.rb]: All classes should start with an uppercase letter

Comment (abconner25) [reply app/models/bishop.rb]: Thanks for the help. I updated all the class names

File: spec/models/chess_piece_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe ChessPiece, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (scottAF) [inline spec/models/chess_piece_spec.rb]: Delete this line so we don't have failures while executing Rspec.

File: app/models/match.rb [hunk 3]
```diff
@@ -5,4 +5,6 @@ class Match < ApplicationRecord
 
     #belongs_to :user
     #uncomment when the user table is created for device
+    has_many :chessPieces
```
Comment (talk2MeGooseman) [inline app/models/match.rb]: This doesnt seem right, have you tested you code your code that you can make a chess piece for a match?

**Pull request merged**